app/com/arpnetworking/play/ProxyClient.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2017 Smartsheet.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.play;
+
+import akka.util.ByteString;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+import org.asynchttpclient.AsyncHandler;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.RequestBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.libs.ws.WSClient;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * A simple proxy client.
+ *
+ * @author Brandon Arp (barp at groupon dot com)
```

Comment:
really? you going back? ;)

Reply:
Uh... hate to say it, but this class really was written while I was at Groupon.  This is copied from Artemis.